core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -0,0 +1,807 @@
+function stageEndPoint(appId) {
+    var words = document.baseURI.split('/');
+    var words2 = document.baseURI.split('?');
+    var ind = words.indexOf("proxy");
+    if (ind > 0) {
+        var appId = words[ind + 1];
+        var stageIdLen = words2[1].indexOf('&');
+        var stageId = words2[1].substr(3, stageIdLen - 3);
+        var newBaseURI = words.slice(0, ind + 2).join('/');
+        return newBaseURI + "/api/v1/applications/" + appId + "/stages/" + stageId;
+    }
+    ind = words.indexOf("history");
+    if (ind > 0) {
+        var appId = words[ind + 1];
+        var attemptId = words[ind + 2];
+        var stageIdLen = words2[1].indexOf('&');
+        var stageId = words2[1].substr(3, stageIdLen - 3);
+        var newBaseURI = words.slice(0, ind).join('/');
+        if (isNaN(attemptId) || attemptId == "0") {
+            return newBaseURI + "/api/v1/applications/" + appId + "/stages/" + stageId;
+        } else {
+            return newBaseURI + "/api/v1/applications/" + appId + "/" + attemptId + "/stages/" + stageId;
+        }
+    }
+    var stageIdLen = words2[1].indexOf('&');
+    var stageId = words2[1].substr(3, stageIdLen - 3);
+    return location.origin + "/api/v1/applications/" + appId + "/stages/" + stageId;
+}
+
+function getColumnNameForTaskMetricSummary(columnKey) {
+    if(columnKey == "executorRunTime") {
+        return "Duration";
+    }
+    else if(columnKey == "jvmGcTime") {
+        return "GC Time";
+    }
+    else if(columnKey == "gettingResultTime") {
+        return "Getting Result Time";
+    }
+    else if(columnKey == "inputMetrics") {
+        return "Input Size / Records";
+    }
+    else if(columnKey == "outputMetrics") {
+        return "Output Size / Records";
+    }
+    else if(columnKey == "peakExecutionMemory") {
+        return "Peak Execution Memory";
+    }
+    else if(columnKey == "resultSerializationTime") {
+        return "Result Serialization Time";
+    }
+    else if(columnKey == "schedulerDelay") {
+        return "Scheduler Delay";
+    }
+    else if(columnKey == "diskBytesSpilled") {
+        return "Shuffle spill (disk)";
+    }
+    else if(columnKey == "memoryBytesSpilled") {
+        return "Shuffle spill (memory)";
+    }
+    else if(columnKey == "shuffleReadMetrics") {
+        return "Shuffle Read Size / Records";
+    }
+    else if(columnKey == "shuffleWriteMetrics") {
+        return "Shuffle Write Size / Records";
+    }
+    else if(columnKey == "executorDeserializeTime") {
+        return "Task Deserialization Time";
+    }
+    return "NA";
+}
+
+$(document).ready(function () {
+    setDataTableDefaults();
+
+    $("#showAdditionalMetrics").append(
+        "<div><a id='taskMetric'>" +
+        "<span class='expand-input-rate-arrow arrow-closed' id='arrowtoggle1'></span>" +
+        " Show Additional Metrics" +
+        "</a></div>" +
+        "<div class='container-fluid' id='toggle-metrics' hidden>" +
```

Comment:
Yes, that is because, each time, we show or hide a row, we have to construct the entire datatable again. Not sure if that is what we want, but we can discuss further.

Reply:
Done